### PR TITLE
PLATO-518: Provide alternative text for the IIIF viewer on the item page

### PR DESCRIPTION
**Pull request by jialin-he**
Resolves PLATO-518

## Description

- Add a description to the IIIF canvas using an aria-label to provide alternative text within the viewer space. 
- We should use title metadata as part of the alternative text if available and fallback to the description if no title is available.

## Checks

**Have you written any necessary unit tests?**

- [ ] Yes
- [ ] Not yet
- [x] Not applicable

**Have you added or updated any necessary integration tests?**

- [ ] Yes
- [x] Not yet
- [ ] Not applicable

**Browsers tested on:**

- [x] Chrome
  - [ ] Mobile
- [ ] Safari
  - [ ] Mobile
- [ ] Firefox
  - [ ] Mobile
- [ ] Edge
- [ ] IE11
- [ ] Not applicable

### CODE REVIEW

**Review by crystalfinch — APPROVED** [review]

File: src/app/asset-page/artstor-viewer/artstor-viewer.component.ts [hunk 1]
```diff
@@ -407,16 +407,19 @@ export class ArtstorViewerComponent implements OnInit, OnDestroy {
...
+
+            // Get all thumbnails in the reference strip
+            let referenceStripThumbnails = document.querySelectorAll(".referencestrip .openseadragon-canvas");
+
+            referenceStripThumbnails.forEach((referenceStripThumbnail, index) => {
+                referenceStripThumbnail.addEventListener("keydown", (event: KeyboardEvent) => {
+                if (event.keyCode === 13) {
+                this.osdViewer.goToPage(index)
+                }
+            })
```
Comment (michael-iden) [inline src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: Formatting here looks a little strange. I think it just needs a few indents.

File: src/app/asset-page/artstor-viewer/artstor-viewer.component.ts [hunk 2]
```diff
@@ -407,16 +407,19 @@ export class ArtstorViewerComponent implements OnInit, OnDestroy {
...
-
-          referenceStripThumbnails.forEach((referenceStripThumbnail, index) => {
-            referenceStripThumbnail.addEventListener("keydown", (event: KeyboardEvent) => {
-             if (event.keyCode === 13) {
-               this.osdViewer.goToPage(index)
-             }
-           })
-          })
+            let mainContentCanvas: HTMLElement = <HTMLElement>document.getElementsByTagName("canvas")[0];
+            mainContentCanvas && mainContentCanvas.setAttribute("aria-label", this.assets[0].title ? this.assets[0].title : this.asset[0].description);
```
Comment (michael-iden) [inline src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: I thought the asset array could be empty when presenting from an image group page. But I could possibly be misremembering. Do you know if this is all good when presenting the group?

Comment (jialin-he) [reply src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: Tested, assets still good when entering from an image group.

File: src/app/asset-page/artstor-viewer/artstor-viewer.component.ts (hunk 1 above)
Comment (jialin-he) [reply src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: Done!

File: src/app/asset-page/artstor-viewer/artstor-viewer.component.ts (hunk 2 above)
Comment (michael-iden) [reply src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: 🎉

File: src/app/asset-page/artstor-viewer/artstor-viewer.component.ts [hunk 3]
```diff
@@ -407,16 +407,19 @@ export class ArtstorViewerComponent implements OnInit, OnDestroy {
...
+
+            // Get all thumbnails in the reference strip
+            let referenceStripThumbnails = document.querySelectorAll(".referencestrip .openseadragon-canvas");
+
+            referenceStripThumbnails.forEach((referenceStripThumbnail, index) => {
+                    referenceStripThumbnail.addEventListener("keydown", (event: KeyboardEvent) => {
+                        if (event.keyCode === 13) {
```
Comment (michael-iden) [inline src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: It still seems a little strange but I think if this line and the one above go back one indent it should be all aligned.

Comment (jialin-he) [reply src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: Ahhh right!

Comment (michael-iden) [reply src/app/asset-page/artstor-viewer/artstor-viewer.component.ts]: 💯

**Review by michael-iden — APPROVED** [review]

**Review by willschoener — APPROVED** [review]

**Pull request merged**